slider/replay.py

```diff
@@ -576,6 +636,62 @@ def parse(cls, data, *, library=None, client=None, save=False):
             **mod_kwargs,
         )
 
+    @lazyval
+    def hits(self):
+        b = self.beatmap
+        rdata = self.actions
+        scores = {"perfects": [],
+                  "goods": [],
+                  "bads": [],
+                  "misses": [],
+                  "sliderbreaks": [],
+                  }
+        hw = od_to_ms(b.od(easy=self.easy, hard_rock=self.hard_rock))
+        rad = circle_radius(b.cs(easy=self.easy, hard_rock=self.hard_rock))
+        i = 0
+        for obj in b.hit_objects:
+            if self.hard_rock:
```

Comment:
hmm i hadn't really thought of doing that. Should i make the `process_slider` and `process_circle` functions static methods as well?

actually might that not cause problems with incrementing `i` (my pointer for the current action)? At the moment it's just incremented whenever i need to but in a separate function that could be tricky